2200070/퍼펙트셔플.py

```diff
@@ -0,0 +1,58 @@
+import sys
+sys.stdin=open('input.txt', 'r')
+
+t=int(input())
+# 테스트 케이스 수
+for _ in range(1, t+1):
+    #문제에서 1번부터 출력하기 요구하므로 1부터 t+1까지의 리스트를 만들고
+    n=int(input())
+    #카드 수를 입력하고
+    d=list(input().split())
+    #카드 구성을 리스트로 받는다
+    d1=[]
+    d2=[]
+    d3=[]
+    #d1, d2는 반반씩 담을 거. d3은 한장씩 뽑을 거
+    for c in range(n):
+        # 여기에서 led(d)를 써야하나 했는데 그냥 n이랑 그게 그거더라
+        if c < n // 2:
+            d1.append(d[c])
+            #c를 인덱스로 하는 리스트 d의 요소를 나눠서 넣을건데
+            #먼저 c가 n을 2로 나눈 것의 몫보다 작으면 d1에 넣어주고
+
+        else:
+            #c가 n을 2로 나눈 몫보다 큰데
+            if n % 2 == 1:
+                #n이 홀수면
+                d1.append(d[c])
+                #d1에 한번 더 넣어주고
+            d2.append(d[c])
+            #나머지 남은 거를 d2에 넣어준다
+            #여기까지 절반 나누기

```

Comment:
현재 풀이는 `n`이 홀수일 때 오답이 나옵니다!
```suggestion
    if n % 2 == 1:
        for c in range(n):
            if c <= n // 2:
                d1.append(d[c])
            else:
                d2.append(d[c])
    else:
        for c in range(n):
            if c < n // 2:
                d1.append(d[c])
            else:
                d2.append(d[c])
```
n이 홀수일 때와 짝수일 때 각각 `d1`과 `d2`를 어떻게 나눠야 하는지 고민해보시면 좋을 것 같습니다 😄

Reply:
리플 달아도 되나요? n이 홀수일 때 오류가 나기 때문에 아래의 while 구문에서 try-except를 써서 해결했습니다. 카드 갯수가 홀수일 경우 d1리스트는 아직 요소가 있는데 d2리스트가 무조건 먼저 바닥나기 때문에 빈 리스트에서 pop할 수 없다고 오류가 나거든요! d2에서 마지막으로 pop 할 때 오류가 나거든 그냥 넘어가거라 하고 명령을 했습니다. 처음부터 if문으로 조건을 달아도 되는군요

Reply:
리플 달아도 되나요? n이 홀수일 때 오류가 나기 때문에 아래의 while 구문에서 try-except를 써서 해결했습니다. 카드 갯수가 홀수일 경우 d1리스트는 아직 요소가 있는데 d2리스트가 무조건 먼저 바닥나기 때문에 빈 리스트에서 pop할 수 없다고 오류가 나거든요! d2에서 마지막으로 pop 할 때 오류가 나거든 그냥 넘어가거라 하고 명령을 했습니다. 처음부터 if문으로 조건을 달아도 되는군요

Reply:
> 리플 달아도 되나요? n이 홀수일 때 오류가 나기 때문에 아래의 while 구문에서 try-except를 써서 해결했습니다. 카드 갯수가 홀수일 경우 d1리스트는 아직 요소가 있는데 d2리스트가 무조건 먼저 바닥나기 때문에 빈 리스트에서 pop할 수 없다고 오류가 나거든요! d2에서 마지막으로 pop 할 때 오류가 나거든 그냥 넘어가거라 하고 명령을 했습니다. 처음부터 if문으로 조건을 달아도 되는군요